source/common/router/upstream_request.cc

```diff
@@ -223,6 +224,20 @@ const Network::Connection& UpstreamRequest::connection() const {
   return *parent_.callbacks()->connection();
 }
 
+void UpstreamRequest::copyDynamicMetaDataFromL4Downstream(StreamInfo::StreamInfo& l4_stream_info) {
+  if (parent_.callbacks()->connection() != nullptr) {
+    const StreamInfo::StreamInfo& downstream_streamInfo =
+        parent_.callbacks()->connection()->streamInfo();
+    const auto& dynamic_metadata = downstream_streamInfo.dynamicMetadata();
+    auto name = Envoy::Extensions::NetworkFilters::NetworkFilterNames::get().DownToUp;
+    auto filter_it = dynamic_metadata.filter_metadata().find(name);
+    if (filter_it != dynamic_metadata.filter_metadata().end()) {
+      ProtobufWkt::Struct metadata(filter_it->second);
```

Comment:
This feels like an unnecessary local copy? Why not pass by reference?

Reply:
downstream and upstream have different lifetime cycle, it is safer to do a local copy.